components/constituency_lookup/ConstituencyLookup.tsx

```diff
@@ -404,23 +405,29 @@ const PostcodeLookup = () => {
 
           {formState.emailOptIn && (
             <>
-              <Form.Control
-                name="email"
-                size="lg"
-                type="email"
-                placeholder="Your Email"
-                value={formState.email}
-                onChange={(e) =>
-                  setFormState({ ...formState, email: e.target.value })
-                }
-                className="my-2 invalid-text-greyed"
-              />
-              {emailError && (
-                <p className="fw-bold fst-italic">
-                  {emailErrorToErrorMessage(emailError)}
-                </p>
-              )}
-
+              <InputGroup hasValidation>
+                <Form.Control
+                  name="email"
+                  size="lg"
+                  type="email"
+                  placeholder="Your Email"
+                  value={formState.email}
+                  isInvalid={emailError ? true : false}
+                  onChange={(e) => {
+                    setFormState({ ...formState, email: e.target.value });
+                    if (!e.target.validity.typeMismatch) {
+                      setEmailError(null);
+                    }
```

Comment:
Not sure if this is the right way do this? Needed otherwise the error hangs around when you've put in a valid email but haven't pressed submit again.

Reply:
There might be other ways to do this, but nothing springing to mind that seems loads better... I think the problem is we only want to show the validation error on submit, rather than as people type, so the standard HTML validation doesn't work, and then we have to do something to remove the validation when the field has been modified.

We could also track last submitted email or just a 'dirty' / 'touched' field, but just seems like extra data to track.

We could alternatively just _always_ set the error to null when the email changes - on the basis that someone is now typing, so hide the error until they re-submit - but I think after they submit, keeping the error until they type a valid email address makes sense, and is more responsive, so think this is good as-is.

Reply:
Yeah absent using a library for validation (which seems like over-kill) I think this is fine.